rskj-core/src/integrationTest/java/co/rsk/snapshotsync/SnapshotSyncIntegrationTest.java

```diff
@@ -0,0 +1,113 @@
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static co.rsk.util.FilesHelper.readBytesFromFile;
+import static co.rsk.util.FilesHelper.writeBytesToFile;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class SnapshotSyncIntegrationTest {
+    private static final String TAG_TO_REPLACE_SERVER_RPC_PORT = "<SERVER_NODE_RPC_PORT>";
+    private static final String TAG_TO_REPLACE_SERVER_PORT = "<SERVER_NODE_PORT>";
+    private static final String TAG_TO_REPLACE_SERVER_DATABASE_PATH = "<SERVER_NODE_DATABASE_PATH>";
+    private static final String TAG_TO_REPLACE_NODE_ID = "<SERVER_NODE_ID>";
+    private static final String TAG_TO_REPLACE_CLIENT_DATABASE_PATH = "<CLIENT_NODE_DATABASE_PATH>";
+    private static final String TAG_TO_REPLACE_CLIENT_PORT = "<CLIENT_PORT>";
+    public static final String TMP_RSKJ_INTEGRATION_TEST_FOLDER = "rskj-integration-test-";
+    private final String rskConfFileServerName = "snap-sync-server-rskj.conf";
+    private final String rskConfFileClientName = "snap-sync-client-rskj.conf";
+    private final int portServer = RandomUtils.nextInt(1, 5000);
+    private final int portClient = RandomUtils.nextInt(5001, 9999);
+    private final String pathDatabaseServer = "/.rsk/regtest/database";
+    private final String randomPathDatabaseClient = RandomStringUtils.randomAlphanumeric(10);
+
+    @Test
+    public void whenStartTheServerAndClientNodes_thenTheClientWillSynchWithServer() throws IOException, InterruptedException {
+        //given
+        Path tempDirectory = Files.createTempDirectory(TMP_RSKJ_INTEGRATION_TEST_FOLDER);
+        Path dirDatabaseServer = Paths.get(System.getProperty("user.home") + pathDatabaseServer);
+        FilesHelper.deleteContents(dirDatabaseServer.toFile());
+        Path tempDirDatabaseClient = Files.createTempDirectory(tempDirectory, randomPathDatabaseClient);
+
+        importTheExportedBlocksToRegtestNode();
+        String rskConfFileChangedServer = configureServerWithGeneratedInformation(dirDatabaseServer);
+        NodeIntegrationTestCommandLine serverNode = new NodeIntegrationTestCommandLine(rskConfFileChangedServer, "--regtest");
+
+        //when
+        serverNode.startNode();
+        ThreadTimerHelper.waitForSeconds(30);
+
+        String rskConfFileChangedClient = configureClientConfWithGeneratedInformation(dirDatabaseServer, tempDirDatabaseClient.toString());
+        NodeIntegrationTestCommandLine clientNode = new NodeIntegrationTestCommandLine(rskConfFileChangedClient, "--regtest");
+        clientNode.startNode();
+
+        ThreadTimerHelper.waitForSeconds(20);
+
+        //then
+        boolean snapshotNotAsserted = true;
+        while (snapshotNotAsserted) {
+            if(clientNode.getOutput().contains("CLIENT - Starting Snapshot sync.") && clientNode.getOutput().contains("CLIENT - Snapshot sync finished!")) {
```

Comment:
What do you mean? Call the getBlockNumber()? I went with this because we had agreed we didn't need to check the data at that moment. Do you have an idea how we could check the database?

Reply:
This was done, check the latest commit @asoto-iov